packages/angular/src/jsonforms-root.component.ts

```diff
@@ -0,0 +1,112 @@
+  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+  THE SOFTWARE.
+*/
+import {
+    Component,
+    EventEmitter, Input, OnChanges, OnInit, Output, SimpleChanges
+} from '@angular/core';
+import { Actions, JsonFormsRendererRegistryEntry, JsonSchema, UISchemaElement, UISchemaTester } from '@jsonforms/core';
+import { JsonFormsAngularService } from './jsonforms.service';
+@Component({
+    selector: 'jsonforms',
+    template: '<jsonforms-outlet></jsonforms-outlet>',
+    providers: [JsonFormsAngularService]
+})
+export class JsonForms implements OnChanges, OnInit {
+
+    @Input() uischema: UISchemaElement;
+    @Input() schema: JsonSchema;
+    @Input() data: any;
+    @Input() renderers: JsonFormsRendererRegistryEntry[];
+    @Input() uischemas: { tester: UISchemaTester; uischema: UISchemaElement; }[];
+    @Output() dataChange = new EventEmitter<any>();
+    @Input() locale: string;
+    @Input() readonly: boolean;
+
+    private initialized = false;
+
+    constructor(private jsonformsService: JsonFormsAngularService) {
+    }
+
+    ngOnInit(): void {
+        this.jsonformsService.init({
+            core: {
+                data: this.data,
+                uischema: this.uischema,
+                schema: this.schema
+            },
+            uischemas: this.uischemas,
+            i18n: { locale: this.locale, localizedSchemas: undefined, localizedUISchemas: undefined },
+            renderers: this.renderers
+        });
+        this.jsonformsService.$state.subscribe(state => {
+            const data = state?.jsonforms?.core?.data;
+            if (data) {
+                this.dataChange.emit(data);
+            }
+        });
+        this.initialized = true;
+    }
+
+    ngOnChanges(changes: SimpleChanges): void {
+        if (!this.initialized) {
+            return;
+        }
+        const newData = changes.data;
+        const newSchema = changes.schema;
+        const newUiSchema = changes.uischema;
+        const newRenderers = changes.renderers;
+        const newUischemas = changes.uischemas;
+        const newLocale = changes.locale;
+        const newReadonly = changes.readonly;
+
+        if (newData && newSchema && newUiSchema) {
+            this.jsonformsService.updateCore(
+                Actions.updateCore(newData.currentValue, newSchema.currentValue, newUiSchema.currentValue)
+            );
+        } else if (newData && !newData.isFirstChange()) {
+            this.jsonformsService.setData(newData.currentValue);
+        } else if (newSchema && !newSchema.isFirstChange()) {
+            this.jsonformsService.setSchema(newSchema.currentValue);
+        } else if (newUiSchema && !newUiSchema.isFirstChange()) {
+            this.jsonformsService.setUiSchema(newUiSchema.currentValue);
+        }
```

Comment:
This will not update the schema when both data and schema change but not the ui schema.

I would do it similar to this:
```typescript
        if (newData || newSchema || newUiSchema) {
            const updateData = newData ? newData.currentValue : this.jsonformsService.$state.core.data;
            let updateSchema = newSchema ? newSchema.currentValue : this.jsonformsService.$state.core.schema;
            let updateUiSchema = newUiSchema ? newUiSchema.currentValue : this.jsonformsService.$state.core.uischema;
            if(updateSchema === undefined) { updateSchema = /* Generate schema */}
            if(updateUiSchema === undefined) { updateUiSchema = /* Generate ui schema */}
            this.jsonformsService.updateCore(
                Actions.updateCore(updateData, updateSchema, updateUiSchema)
            );
        }
```